Uni_life_tracker/app/src/main/java/Database/EventDatabaseHelper.java

```diff
@@ -88,18 +89,59 @@ public void deleteEvent(Event event){
         try {
             cur = db.query("events", null, null, null,
                     null, null, null, null);
-            if (cur != null && cur.moveToFirst()) {
-                do {
-                    LocalDateTime start = LocalDateTime.parse(cur.getString(cur.getColumnIndexOrThrow("START_DATE")), DATEFORMAT);
-                    LocalDateTime end = LocalDateTime.parse(cur.getString(cur.getColumnIndexOrThrow("END_DATE")), DATEFORMAT);
-                    Event event = new Event(cur.getString(cur.getColumnIndexOrThrow("NAME")),
-                                            cur.getInt(cur.getColumnIndexOrThrow("PRIORITY")),
-                                            start,
-                                            end);
-                    eventList.add(event);
-
-                } while (cur.moveToNext());
+            if (cur != null) {
+                if (cur.moveToFirst()) {
+                    do {
+                        LocalDateTime start = LocalDateTime.parse(cur.getString(cur.getColumnIndexOrThrow("START_DATE")), DATEFORMAT);
+                        LocalDateTime end = LocalDateTime.parse(cur.getString(cur.getColumnIndexOrThrow("END_DATE")), DATEFORMAT);
+                        Event event = new Event(cur.getString(cur.getColumnIndexOrThrow("NAME")),
+                                                cur.getInt(cur.getColumnIndexOrThrow("PRIORITY")),
+                                                start,
+                                                end);
+                        eventList.add(event);
+
+                    } while (cur.moveToNext());
+
+                }
+            }
+        } finally {
+            db.endTransaction();
+            assert cur != null;
+            cur.close();
+        }
 
+        return eventList;
+    }
+
+    /**
+     * Return a list of all events which start today currently in the database
+     *
+     * @return returns the list of Event objects which start today
+     */
+    public List<Event> getTodaysEvents(){
+        List<Event> eventList = new ArrayList<>();
+        Cursor cur = null;
+        db.beginTransaction();
+        try {
+            cur = db.query("events", null, null, null,
```

Comment:
Is there a reason why all these are all null?

Reply:
This is basically a 'SELECT * FROM events' query. I can rewrite this using a different method that shows what the query is if that helps make things clearer.